### ISSUE TITLE
Gzip data event

### ISSUE THREAD

**Issue by alexcroox** [IndigoUnited/node-request-progress#20]
In the normal request library with `gzip: true` the 'data' event is the decompressed data, making data.length accurate to the original size as it is on the server, rather than the gzip size. Is this still true for `progress.on('data')?`

```
).on('data', function(data) {
    // decompressed data as it is received
    console.log('decoded chunk: ' + data)
  })
```

**Comment by alexcroox** [comment]
The reported download speed is also 2/3 MB/s more than my connection is capable of. I've directly compared it several times to a build that is identical except for gzip.

**Comment by satazor** [comment]
The data event is the decompressed one so I guess you are right. I don't know if its possible to get the raw events instead. I'm on vacations atm so I can't investigate.

**Comment by satazor** [comment]
@alexcroox If not possible to get the raw data, we could decrement a % of the data size with the average compression ratio that gzip offers.

**Comment by alexcroox** [comment]
Well in the `request` library the `data` event exposes raw data when gzip is enabled which is why I was originally asking to know if this library simply proxied that event and it would be the same or not (I'm not sure how I can test it myself).

![ss](http://cl.ly/gB0Y/d)

https://github.com/request/request#user-content-examples

**Comment by satazor** [comment]
So if I understand correctly we must listen to `response` event to get the response, and then listen to its data events to get the raw data. That would be a straightforward change I think: we must change this [listener](https://github.com/IndigoUnited/node-request-progress/blob/master/index.js#L117) to the [response](https://github.com/IndigoUnited/node-request-progress/blob/master/index.js#L15) object.

Feel free to make the changes to see if it solves the problem. If it does, please send a PR otherwise I can only fix this in the end of my vacations.

**Comment by pedrorfernandes** [comment]
By forcing a `reportState()` when we reach `onEnd()` I can confirm the issue:

```javascript
// in master, with gzip: true
Progress: { time: { elapsed: 0.02, remaining: null },
  speed: null,
  percentage: 1,
  size: { total: 14984, transferred: 45201 } }
```
But with the fix suggested by @alexcroox and @satazor:
```javascript
.on('response', function handleResponse(response) {
    response.on('data', onData.bind(null, context));
    return onResponse(context, response);
})
```

We get:
```javascript
// with suggested by @alexcroox and @satazor, with gzip: true
Progress: { time: { elapsed: 0.016, remaining: null },
  speed: null,
  percentage: 1,
  size: { total: 14960, transferred: 14960 } }
```

So it seems to work! I will open a PR soon with this change

**Comment by satazor** [comment]
@pedrorfernandes Looks good! Additionally could you also make a PR for #24? It's just a naming change and since I'm going to do a major bump, it makes sense to aggregate both changes in the same major version.

**Comment by pedrorfernandes** [comment]
@satazor roger that, will do!

**Comment by satazor** [comment]
@alexcroox Support for gzip released as v3.0.0.

**Comment by alexcroox** [comment]
Great work guys, thank you so much!
